src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -172,8 +172,18 @@ public static function get_core_data() {
 		 *
 		 * @param WP_Theme_JSON_Data $theme_json Class to access and update the underlying data.
 		 */
-		$theme_json   = apply_filters( 'wp_theme_json_data_default', new WP_Theme_JSON_Data( $config, 'default' ) );
-		static::$core = $theme_json->get_theme_json();
+		$theme_json = apply_filters( 'wp_theme_json_data_default', new WP_Theme_JSON_Data( $config, 'default' ) );
```

Comment:
Hi, is anyone able to reproduce this error?

I cannot repro the issue reported using latest WordPress `trunk` and Gutenberg 14.8.1 (the last release prior to updating `WP_Theme_JSON_Data_Gutenberg`). I'd like to understand this better before jumping into implementation.

The only way I can think to reproduce the issue is by not using the provided structure to update the data as a consumer of this filter. This is, instead of doing:

```php
add_filter( 'wp_theme_json_data_*', function( $theme) {
  $new_data = array( /* ... */ );

  return $theme->update_with( $new_data );
});
```

Your plugin does something like:

```php
add_filter( 'wp_theme_json_data_*', function( $theme) {
  /* your plugin code */
  return  $your_custom_structure;
});

```

Note that this is not supported or recommended. The `update_with` method is a requirement to use these filters, as [stated in the devnote](https://make.wordpress.org/core/2022/10/10/filters-for-theme-json-data/), because it makes sure everything works correctly (data migration, sanitization, etc.). This is [what happened](https://core.trac.wordpress.org/ticket/61112#comment:19) to @ryelle, as she reported.

Is there any other use case?

Reply:
To be clear, I'm not _against_ doing it, but I'm _reluctant_ to given that it only happens if a plugin provides their own structure, which is not supported or recommended.

Reply:
The issue can be reproduced using the [Blocks Everywhere](https://wordpress.org/plugins/blocks-everywhere/) plugin, which does the following, basically:

```php
add_filter( 'wp_theme_json_data_theme', function( $theme) {
	$theme = new \WP_Theme_JSON_Data_Gutenberg( /* ... */ );
	return $theme;
} );
```

Which I suppose is incorrect according to the dev note, but it did work before the recent change.

Reply:
@oandregal I believe this was only reproducible prior to `WP_Theme_JSON_Data_Gutenberg::get_theme_json` being added in version 18.3.0, which shipped after the core change. If we don't need to worry about WP 6.6 compat with any Gutenberg versions prior to 18.3.0, then we can probably close this issue as `wontfix`.

However, I think this use case is instructive for us as we consider better ways of managing the `WP_Theme_JSON_` classes between the GB repo and the WP-dev repo to avoid these types of conflicts.